com.unity.multiplayer.mlapi/Runtime/Core/BatchingUtil.cs

```diff
@@ -0,0 +1,187 @@
+using MLAPI.Serialization.Pooled;
+using MLAPI.Serialization;
+using MLAPI.Configuration;
+using MLAPI.Profiling;
+using System.Collections.Generic;
+using System.Linq;
+using System;
+using System.IO;
+
+namespace MLAPI
+{
+    class BatchUtil
+    {
+        public class SendStream
+        {
+            public FrameQueueItem Item;
+            public PooledBitWriter Writer;
+            public PooledBitStream Stream = new PooledBitStream();
+        }
+
+        // Stores the stream of batched RPC to send to each client, by ClientId
+        private Dictionary<ulong, SendStream> SendDict = new Dictionary<ulong, SendStream>();
+
+        public void PushLength(int length, ref PooledBitWriter writer)
+        {
+            // supports lengths up to 16-bit wide
+            if (length < 255)
+            {
+                writer.WriteByte((byte)length); // write the amounts of bytes that are coming up
+            }
+            else
+            {
+                writer.WriteByte(255); // mark larger size
+                writer.WriteByte((byte)(length % 256)); // write the length modulo 256
+                writer.WriteByte((byte)(length / 256)); // write the length divided by 256
+            }
+        }
+
+        public int PopLength(in BitStream messageStream)
+        {
+            int length = 0;
+            byte read = (byte)messageStream.ReadByte();
+            if (read != 255)
+            {
+                return read;
+            }
+            length += messageStream.ReadByte();
+            length += messageStream.ReadByte() * 256;
+
+            return length;
+        }
+
+        /// <summary>
+        /// GetTargetList
+        /// Returns the list of ClientId an item is targeted to
+        /// </summary>
+        /// <param name="item">the FrameQueueItem we want targets for</param>
+        private static List<ulong> GetTargetList(in FrameQueueItem item)
+        {
+            List<ulong> ret = new List<ulong>();
+            switch (item.QueueItemType)
+            {
+                case RPCQueueManager.QueueItemType.ServerRPC:
+                    ret.Add(item.NetworkId);
```

Comment:
What if you move the allocation to only happen in this case statement since it's skipped in the other 2?

Reply:
Allocation is gone now. :-)